### PR TITLE
Add rpaas debug feature with custom image

### PR DESCRIPTION
**Pull request by morpheu**

### CODE REVIEW

File: internal/pkg/rpaas/k8s.go [hunk 1]
```diff
@@ -133,44 +142,171 @@ func (q *fixedSizeQueue) Next() *remotecommand.TerminalSize {
...
-func (m *k8sRpaasManager) Exec(ctx context.Context, instanceName string, args ExecArgs) error {
-	instance, err := m.GetInstance(ctx, instanceName)
+func (m *k8sRpaasManager) Debug(ctx context.Context, instanceName string, args DebugArgs) error {
+	instance, debugContainerName, status, err := m.debugPodWithContainerStatus(ctx, &args, instanceName)
 	if err != nil {
 		return err
 	}
+	if status.State.Terminated != nil {
+		req := m.kcs.CoreV1().Pods(instance.Namespace).GetLogs(args.Pod, &corev1.PodLogOptions{Container: debugContainerName})
+		return logs.DefaultConsumeRequest(req, args.Stdout)
```
Comment (nettoclaudio) [inline internal/pkg/rpaas/k8s.go:152]: It's possible I overlooked something however I really didn't get it... Why should we see the logs if pod is the termination phase?

File: internal/pkg/rpaas/k8s.go [hunk 2]
```diff
@@ -133,44 +142,171 @@ func (q *fixedSizeQueue) Next() *remotecommand.TerminalSize {
...
 	}
+	debugPatch, err := strategicpatch.CreateTwoWayMergePatch(podJS, podWithDebugJS, instancePod)
+	if err != nil {
+		return err
+	}
+	err = m.cli.SubResource("ephemeralcontainers").Patch(ctx, &instancePod, client.RawPatch(types.StrategicMergePatchType, debugPatch))
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (nettoclaudio) [inline internal/pkg/rpaas/k8s.go:245]:
```suggestion
	return m.cli.SubResource("ephemeralcontainers").Patch(ctx, &instancePod, client.RawPatch(types.StrategicMergePatchType, debugPatch))
```

File: pkg/web/transport.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
+package web
```
Comment (nettoclaudio) [inline pkg/web/transport.go:1]: Missing license header.

**Review by nettoclaudio — APPROVED** [review]
Some in-line request changes 💅  but LGTM

File: internal/pkg/rpaas/k8s.go (hunk 2 above)
Comment (morpheu) [reply internal/pkg/rpaas/k8s.go:244]: nice catch :)

File: internal/pkg/rpaas/k8s.go (hunk 1 above)
Comment (morpheu) [reply internal/pkg/rpaas/k8s.go:152]: Cause it's possible to send a command and not enter in interact mode with debug container. In this case scenario container enter in terminated state and code throw to user container logs related to the command sent (this feature its similar how kubectl debug works in this scenario)

File: cmd/plugin/rpaasv2/cmd/debug.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+		done := make(chan error, 1)
+		go func() {
+			defer close(done)
+			for {
+				mtype, message, nerr := conn.ReadMessage()
+				if nerr != nil {
+					closeErr, ok := nerr.(*websocket.CloseError)
+					if !ok {
+						done <- fmt.Errorf("ERROR: receveid an unexpected error while reading messages: %w", err)
```
Comment (crgisch) [inline cmd/plugin/rpaasv2/cmd/debug.go:113]: Small typo: 
"ERROR: _received_ an unexpected..."

**Review by crgisch — APPROVED** [review]
LGTM

File: cmd/plugin/rpaasv2/cmd/debug.go (hunk 4 above)
Comment (morpheu) [reply cmd/plugin/rpaasv2/cmd/debug.go:113]: 🦅  eyes :) 
thanks!

**Pull request merged**